### PR TITLE
Gen kw comments

### PR DESCRIPTION
**Pull request by joakim-hove**
**Task**
1. Add support for comments in GEN_KW parameter files.
2. Add support for `TRIANGULAR` prior distribution.
3. Refactored transformat function implementation.

**Approach**
Used the standard config parser for parsing the GEN_KW parameter files, have allowed the config parser to dynamically add unrecognized keywords.

**Pre un-WIP checklist**
- [x] Statoil tests pass locally
- [ ] Have completed graphical integration test steps

### CODE REVIEW

File: libconfig/src/config_parser.c [hunk 1]
```diff
@@ -447,35 +447,48 @@ bool config_parser_add_key_values(config_parser_type * config,
...
   if (!config_has_schema_item(config, kw)) {
-    if (unrecognized == CONFIG_UNRECOGNIZED_WARN)
-      fprintf(stderr,
-              "** Warning keyword: %s not recognized when parsing: %s --- \n",
+
+    if (unrecognized == CONFIG_UNRECOGNIZED_IGNORE)
+      return false;
+
+    if (unrecognized == CONFIG_UNRECOGNIZED_WARN) {
+      fprintf(stderr, "** Warning keyword: %s not recognized when parsing: %s --- \n",
```
Comment (markusdregi) [inline libconfig/src/config_parser.c:456]: Should we log this?

File: libenkf/src/gen_kw_config.c [hunk 2]
```diff
@@ -129,24 +132,44 @@ void gen_kw_config_set_parameter_file( gen_kw_config_type * config , const char
...
+                                                 CONFIG_UNRECOGNIZED_ADD,
+                                                 false);
+    for (int item_index = 0; item_index < config_content_get_size( content); item_index++ ){
+      const config_content_node_type * node = config_content_iget_node(content, item_index);
+      const char * parameter_name = config_content_node_get_kw(node);
+      gen_kw_parameter_type * parameter = gen_kw_parameter_alloc(parameter_name, config->tag_fmt);
+      trans_func_type * trans_func = trans_func_alloc(config_content_node_get_stringlist(node));
+      gen_kw_parameter_set_trans_func(parameter, trans_func);
+      vector_append_owned_ref( config->parameters , parameter , gen_kw_parameter_free__ );
+      printf("Looking at key:%s\n",parameter_name);
```
Comment (markusdregi) [inline libenkf/src/gen_kw_config.c]: Remove

File: libenkf/src/gen_kw_config.c [hunk 3]
```diff
@@ -129,24 +132,44 @@ void gen_kw_config_set_parameter_file( gen_kw_config_type * config , const char
...
-      fscanf_return = fscanf(stream , "%s" , parameter_name);
-      if (fscanf_return == 1) {
-        gen_kw_parameter_type * parameter  = gen_kw_parameter_alloc( parameter_name , config->tag_fmt);
-        trans_func_type       * trans_func = trans_func_fscanf_alloc( stream, parameter_file );
-        gen_kw_parameter_set_trans_func( parameter , trans_func );
-
-        vector_append_owned_ref( config->parameters , parameter , gen_kw_parameter_free__ );
-      } else
-        break; /* OK - we are ate EOF. */
+    /* FILE * stream = util_fopen(parameter_file , "r"); */
```
Comment (markusdregi) [inline libenkf/src/gen_kw_config.c]: Please remove the commented code

File: libenkf/src/gen_kw_config.c (hunk 2 above)
Comment (markusdregi) [reply libenkf/src/gen_kw_config.c]: Never mind, was removed in a later commit.

File: libenkf/src/gen_kw_config.c (hunk 3 above)
Comment (markusdregi) [reply libenkf/src/gen_kw_config.c]: Never mind, was removed in a later commit.

File: libconfig/src/config_parser.c (hunk 1 above)
Comment (joakim-hove) [reply libconfig/src/config_parser.c:456]: The config parser is also used for other tasks where logging to the shared log facility does not make sense  - so I would say no?

Comment (markusdregi) [reply libconfig/src/config_parser.c:456]: You say no, I say merge.

**Review by markusdregi — APPROVED** [review]
LGTM!

**Pull request merged**